### PR TITLE
🌱 Added Error message when reconciling loop is triggered more than once

### PR DESCRIPTION
**Pull request by babugeet**
**What this PR does / why we need it**:
Added Error message when reconciling loop is triggered more than once

Fixes #
https://github.com/metal3-io/baremetal-operator/issues/1787#event-13218719410

### CODE REVIEW

**Comment by tuminoid** [comment]
Please sign-off the work for DCO, and fix the linter nits.

**Comment by babugeet** [comment]
@tuminoid : lint issues and DCO is fixed.. pls check

File: controllers/metal3.io/baremetalhost_controller.go [hunk 1]
```diff
@@ -232,7 +236,7 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 
 	// Only save status when we're told to, otherwise we
 	// introduce an infinite loop reconciling the same object over and
-	// over when there is an unrecoverable error (tracked through the
+	// over when there is an unrecoveràble error (tracked through the
```
Comment (tuminoid) [inline controllers/metal3.io/baremetalhost_controller.go:239]: Some weird accent here for no reason

Comment (babugeet) [reply controllers/metal3.io/baremetalhost_controller.go]: fixed, but not sure how it came though

File: controllers/metal3.io/baremetalhost_controller.go [hunk 2]
```diff
@@ -212,12 +213,15 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
...
+	if err != nil || !ready {
+		var msg string
+		if err == nil {
+			msg = NotReady
+		} else {
+			msg = err.Error()
+		}
 		provisionerNotReady.Inc()
-		reqLogger.Info("provisioner is not ready", "RequeueAfter:", provisionerNotReadyRetryDelay)
+		reqLogger.Info("provisioner is not ready", "Error", msg, "RequeueAfter:", provisionerNotReadyRetryDelay)
```
Comment (mquhuy) [inline controllers/metal3.io/baremetalhost_controller.go:224]: "Error:"

File: controllers/metal3.io/bmceventsubscription_controller.go [hunk 3]
```diff
@@ -100,12 +100,14 @@ func (r *BMCEventSubscriptionReconciler) Reconcile(ctx context.Context, request
...
-	if !ready {
-		reqLogger.Info("provisioner is not ready", "RequeueAfter:", provisionerNotReadyRetryDelay)
+	if err != nil || !ready {
+		var msg string
+		if err == nil {
+			msg = NotReady
+		} else {
+			msg = err.Error()
+		}
+		reqLogger.Info("provisioner is not ready", "Error", msg, "RequeueAfter:", provisionerNotReadyRetryDelay)
```
Comment (mquhuy) [inline controllers/metal3.io/bmceventsubscription_controller.go:110]: "Error:"

File: controllers/metal3.io/bmceventsubscription_controller.go [hunk 4]
```diff
@@ -220,12 +222,14 @@ func (r *BMCEventSubscriptionReconciler) getProvisioner(ctx context.Context, req
...
```
Comment (mquhuy) [inline controllers/metal3.io/bmceventsubscription_controller.go:232]: "Error:"

File: controllers/metal3.io/baremetalhost_controller.go [hunk 5]
```diff
@@ -212,12 +213,15 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 	}
 
 	ready, err := prov.TryInit()
-	if err != nil {
-		return ctrl.Result{}, errors.Wrap(err, "failed to check services availability")
-	}
-	if !ready {
+	if err != nil || !ready {
```
Comment (mquhuy) [inline controllers/metal3.io/baremetalhost_controller.go:216]: Log-wise, how is it different from just return an error if err != nil?

**Review by mquhuy — COMMENTED** [review]
Some nits.

File: controllers/metal3.io/baremetalhost_controller.go (hunk 5 above)
Comment (babugeet) [reply controllers/metal3.io/baremetalhost_controller.go:216]: in some of the implementation of TryInit(), it was returning ready state as bool 'false' and at the same time err was nil. Also this is the standard followed in the code base

File: controllers/metal3.io/baremetalhost_controller.go (hunk 2 above)
Comment (babugeet) [reply controllers/metal3.io/baremetalhost_controller.go]: fixed

File: controllers/metal3.io/bmceventsubscription_controller.go (hunk 3 above)
Comment (babugeet) [reply controllers/metal3.io/bmceventsubscription_controller.go]: fixed

**Review by Rozzii — COMMENTED** [review]
/approve

File: controllers/metal3.io/baremetalhost_controller.go (hunk 2 above)
Comment (dtantsur) [reply controllers/metal3.io/baremetalhost_controller.go]: This was wrong btw, please don't add colons to structured logging

**Pull request merged**